docs/framework/react/typescript.md

```diff
@@ -239,6 +239,26 @@ const data = queryClient.getQueryData<Group[]>(['groups'])
 [//]: # 'TypingQueryOptions'
 [//]: # 'Materials'
 
+## Typing Mutation Options
+
+Similarly to `queryOptions`, you can use `mutationOptions` to extract mutation options into a separate function:
+
+```ts
+function groupMutationOptions() {
+  return mutationOptions({
+    mutationKey: ['groups'], // mutationKey is required
+    mutationFn: addGroup,
+  })
+}
+
+useMutation({
+  ...groupMutationOptions()
+  onSuccess: () => queryClient.invalidateQueries({ queryKey: ['groups'] })
```

Comment:
I think this might be a bit misleading, as the `mutationKey` and the `queryKey` are the same here, so it could seem like they are related / have to be the same, but that’s not true.